quadriga/src/main/java/edu/asu/diging/quadriga/web/ListCollectionController.java

```diff
@@ -1,37 +1,79 @@
 package edu.asu.diging.quadriga.web;
 
-import javax.servlet.http.HttpServletRequest;
 
+
+
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.domain.PageRequest;
+import org.springframework.security.core.Authentication;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
 
-import edu.asu.diging.quadriga.core.data.CollectionRepository;
+import edu.asu.diging.quadriga.core.model.users.SimpleUserApp;
+import edu.asu.diging.quadriga.core.service.CollectionManager;
+import edu.asu.diging.quadriga.core.service.SimpleUserAppService;
+import edu.asu.diging.simpleusers.core.model.impl.SimpleUser;
 
 @Controller
 public class ListCollectionController {
 
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    
+    @Autowired
+    private CollectionManager collectionManager;
+    
     @Autowired
-    private CollectionRepository collectionRepo;
+    public SimpleUserAppService simpleUserAppService;
+    
 
     @RequestMapping(value = "/auth/collections", method = RequestMethod.GET)
-    public String list(HttpServletRequest request, Model model) {
+    public String list(@RequestParam(defaultValue = "1", required = false, value = "page") String page,
+            @RequestParam(defaultValue = "20", required = false, value = "size") String size, Model model, Authentication authentication) {
 
-        int page = 0;
-        int size = 20;
+        SimpleUser simpleUser = (SimpleUser) authentication.getPrincipal();
 
-        if (request.getParameter("page") != null && !request.getParameter("page").isEmpty()) {
-            page = Integer.parseInt(request.getParameter("page")) - 1;
+        List<SimpleUserApp> userApps = simpleUserAppService.findByUsername(simpleUser.getUsername());
```

Comment:
you probably want to paginate here since if there are many collections for a user, you don't want to put all the apps here into memory.

Reply:
?

Reply:
I see, pagination doesn't make too much sense here. but I would get only the ids directly from the database, not the who app objects if it's unnecessary.